player/Player.cs

```diff
@@ -33,7 +34,21 @@ public override void _PhysicsProcess(float delta)
             Input.GetActionStrength("ui_right") - Input.GetActionStrength("ui_left"),
             Input.GetActionStrength("ui_down") - Input.GetActionStrength("ui_up")
         );
-        var velocity = input.Normalized();
+        var velocity = input.Normalized();
+
+        if (hitsQueued > 0)
+        {
+            animationTree.Set("parameters/Hit/blend_position", velocity);
+            animationStateMachine.Start("Hit");
+
+            // Deduct health.
+            GetNode<Inventory>("/root/Game/UI/Inventory").DeductHealth(hitsQueued);
+
+            hitsQueued = 0;
+            return;
+        }
+
+        if (animationStateMachine.GetCurrentNode().Equals("Hit")) return;
```

Comment:
if we want to wait for the animation to finish, we should use the `await ToSignal()` syntax instead of returning